mod_ci/controllers.py

```diff
@@ -866,6 +868,57 @@ def show_maintenance():
     }
 
 
+@mod_ci.route('/blocked_users')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer()
+def blocked_users():
+        blocked_users = BlockedUsers.query.order_by(BlockedUsers.userID)
+
+        usernames = {}  # Initialize usernames dictionary
+        for instance in blocked_users:
+            # Set values as an error variable
+            usernames[instance] = 'Error, cannot get username'
+        for key in usernames.keys():
+            # Fetch usernames from GitHub API
+            api_url = requests.get('https://api.github.com/user/{}'.format(key), timeout=30)
+            if requests.exceptions.RequestException is not None:
+                break
+            userdata = api_url.json()
+            # Set values to the actual usernames if no errors
+            usernames[key] = userdata['login']
+
+        # Define addUserForm processing
+        addUserForm = AddUsersToBlacklist()
+        if addUserForm.validate_on_submit():
+            blocked_user = BlockedUsers(addUserForm.userID.data, addUserForm.comment.data)
+            if BlockedUsers.query.filter_by(userID=addUserForm.userID.data).first() is not None:
+                flash('User already blocked.')
+                return redirect(url_for('.blocked_users'))
+            g.db.add(blocked_user)
+            g.db.commit()
+            flash('User blocked successfully.')
+            return redirect(url_for('.blocked_users'))
+
+        # Define removeUserForm processing
+        removeUserForm = RemoveUsersFromBlacklist()
+        if removeUserForm.validate_on_submit():
+            blocked_user = BlockedUsers.query.filter_by(userID=removeUserForm.userID.data).first()
+            if blocked_user is None:
+                flash('No such user in Blacklist')
+                return redirect(url_for('.blocked_users'))
+            g.db.remove(blocked_user)
+            g.db.commit()
+            flash('User removed successfully.')
+            return redirect(url_for('.blocked_users'))
+
+        return{
+            'addUserForm': addUserForm(),
```

Comment:
Why create a new form if you already defined it earlier? This way you're not retaining any errors the form might hold.